kv/rest_test.go

```diff
@@ -508,11 +510,15 @@ func getURL(url string, t *testing.T) string {
 }
 
 func httpDo(addr, method, path string, body io.Reader) (*http.Response, error) {
-	req, err := http.NewRequest(method, "http://"+addr+path, body)
+	req, err := http.NewRequest(method, "https://"+addr+path, body)
 	if err != nil {
 		return nil, err
 	}
-	return http.DefaultClient.Do(req)
+	return httpDoReq(req)
+}
+
+func httpDoReq(req *http.Request) (*http.Response, error) {
+	return client.CreateTestHTTPClient().Do(req)
```

Comment:
True. A next step of this is to refactor the http client code. Right now, it's all over the place.
At the very least, tls configs should be persistent, but the only place right now is context which does not have room for both client and server configs. Basically, the whole thing needs refactoring.